### ISSUE TITLE
rustdoc rendered `const`s should respect `#[doc(hidden)]` fields

### ISSUE THREAD

**Issue by SergioBenitez** [rust-lang/rust#49259]
For a `struct` and `const` declared as:

```rust
pub struct HiddenFoo {
    #[doc(hidden)]
    pub foo: usize,
}

pub const SOME_FOO: HiddenFoo = HiddenFoo { foo: 10 };
```

`rustdoc` currently renders:

> pub const SOME_FOO: HiddenFoo
> ```rust
> SOME_FOO: HiddenFoo = HiddenFoo { foo: 10 };
> ```

This _unhides_ the hidden `foo` field. Instead, it should respect the `#[doc(hidden)]` attribute on `foo` and render the following:

> pub const SOME_FOO: HiddenFoo
> ```rust
> SOME_FOO: HiddenFoo = HiddenFoo { .. };
> ```

Or, better yet, if all fields are hidden, then I'd prefer for it to not render the definition at all:

> pub const SOME_FOO: HiddenFoo

**Comment by GuillaumeGomez** [comment]
Seems complicated since at the point we print it, we don't have fields declaration attributes anymore...

**Comment by ollie27** [comment]
Fixed by #53409. We don't show the values of consts anymore.
